recipes-connectivity/bluetooth/files/broadcom-bluetooth/update-brcm-btfw-load-service.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+
+FIRMWARE="/system/etc/firmware"
+
+found=0
+MMC=("mmc0" " mmc1" " mmc2")
+for mmc in ${MMC[@]}
+do
+    if [[ -f /sys/bus/sdio/devices/${mmc}:0001:1/vendor ]] && [[ -f /sys/bus/sdio/devices/${mmc}:0001:1/device ]]; then
+        VENDOR=$(cat /sys/bus/sdio/devices/${mmc}:0001:1/vendor)
+        DEVICE=$(cat /sys/bus/sdio/devices/${mmc}:0001:1/device)
+        found=1
+        break
+    fi
+done
+
+if [[ "$found" == "0" ]]; then
+    echo "Do not need to load Bluetooth firmware."
+    echo "Exit."
+    exit 0
+fi
+
+echo "VENDOR=$VENDOR"
+echo "DEVICE=$DEVICE"
+
+BT_UART="$(strings /sys/firmware/fdt | grep bt_uart)"
+if [[ -n ${BT_UART} ]]; then
+    NUMBER=$(echo "${BT_UART}" | cut -d "_" -f2 | cut -d "t" -f2)
+    BT_UART="/dev/ttyS${NUMBER}"
+else
+    exit 0
+fi
+echo "BT_UART: ${BT_UART}"
+
+if [[ "$VENDOR" == "0x02d0" ]]; then
+    # ap6236
+    if [[ "$DEVICE" == "0xa9a6" ]]; then
+        FIRMWARE="$FIRMWARE/BCM4343B0.hcd"
+        echo "load ap6236 bt firmware"
+    # ap6256
+    elif [[ "$DEVICE" == "0xa9bf" ]]; then
+        FIRMWARE="$FIRMWARE/BCM4345C5.hcd"
+        echo "load ap6256 bt firmware"
+    # ap6398s
+    elif [[ "$DEVICE" == "0x4359" ]]; then
+        FIRMWARE="$FIRMWARE/BCM4359C0.hcd"
+        echo "load ap6398s bt firmware"
+    fi
+fi
+
```

Comment:
Do we need to update brcm-btfw-update.service each time? The content is the same so not clear why we need to do this.

Reply:
Each time, the script, update-brcm-btfw-load-service.sh, is executed to update the file, /lib/systemd/system/brcm-btfw-load.service. Once done, the service broadcom-btfw-load.service will start to load the BT firmware.

Reply:
Why do we need to update it each time? Is anything changing after each boot?